apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js

```diff
@@ -0,0 +1,86 @@
+import {
+  MICROBIT_FIRMATA_V1_URL,
+  MICROBIT_FIRMATA_V2_URL,
+  MICROBIT_IDS_V1,
+  MICROBIT_IDS_V2,
+  MICROBIT_V1,
+  MICROBIT_V2
+} from '@cdo/apps/lib/kits/maker/boards/microBit/MicroBitConstants';
+import {DAPLink, WebUSB} from 'dapjs';
+import {getStore} from '@cdo/apps/redux';
+import {setMicrobitFirmataUpdatePercent} from '@cdo/apps/lib/kits/maker/microbitRedux';
+import {EventEmitter} from 'events'; // provided by webpack's node-libs-browser
+
+export default class MBFirmataUpdater extends EventEmitter {
+  constructor() {
+    super();
+    // State value 'microbitFirmataUpdatePercent' is only updated when this instance variable changes.
+    this.firmataUpdatePercent = 0;
+  }
+
+  detectMicrobitVersion(device) {
+    // Detect micro:bit version and select the right Intel Hex for micro:bit V1 or V2
+    const microbitId = device.serialNumber.substring(0, 4);
+    let microbitVersion = MICROBIT_V1;
+    if (MICROBIT_IDS_V2.includes(microbitId)) {
```

Comment:
Would it be clearer to have `null` be the default case?

Either...
```
let microbitVersion = null;
if (MICROBIT_IDS_V2.includes(microbitId)) {
  microbitVersion = MICROBIT_V2;
} else if (MICROBIT_IDS_V1.includes(microbitId)) {
  microbitVersion = MICROBIT_v1;
}
return microbitVersion;
```

or, skip the variable altogether and do:

```
if (MICROBIT_IDS_V2.includes(microbitId)) {
  return MICROBIT_V2;
}
if (MICROBIT_IDS_V1.includes(microbitId)) {
  return MICROBIT_v1;
}
return null;
}

Reply:
Sure! I like your first suggestion - similar to Sanchit's feedback for assigning URL - thanks!